proposals/bulk-memory-operations/Overview.md

```diff
@@ -272,43 +286,24 @@ The instruction has the signature `[i32 i32 i32] -> []`. The parameters are, in
 
 It is a validation error to use `memory.init` with an out-of-bounds segment index.
 
-A trap occurs if:
-* the segment is used after it has been dropped via `data.drop`. This includes
-  active segments that were dropped after being copied into memory during module
-  instantiation.
-* any of the accessed bytes lies outside the source data segment or the target memory
+A trap occurs if any of the accessed bytes lies outside the source data segment
+or the target memory.  Bounds checking is performed before any data are written.
+The order of writing is unspecified.
```

Comment:
It's worth noting that this is not currently observable, so does not compromise determinism.